mne_bids/mne_bids.py

```diff
@@ -528,56 +514,31 @@ def _sidecar_json(raw, task, manufacturer, fname, kind, eeg_ref=None,
     return fname
 
 
-def raw_to_bids(subject_id, task, raw_file, output_path, session_id=None,
-                acquisition=None, run=None, kind='meg', events_data=None,
-                event_id=None, hpi=None, electrode=None, hsp=None,
-                eeg_ref=None, eeg_gnd=None, config=None,
-                overwrite=False, verbose=True):
+def write_raw_bids(raw, bids_fname, output_path, events_data=None,
+                   event_id=None, overwrite=False, verbose=True):
     """Walk over a folder of files and create BIDS compatible folder.
 
     Parameters
     ----------
-    subject_id : str
-        The subject name in BIDS compatible format ('01', '02', etc.)
-    task : str
-        Name of the task the data is based on.
-    raw_file : str | instance of mne.Raw
+    raw : instance of mne.Raw
```

Comment:
+1 ... we do have some special code in our `read_raw` that can help users to not run into errors

Reply:
I don't want to necessarily advocate strongly for that approach of using `read_raw`. There have been many discussions in the past on the MNE github list about the possibility of a generalized reader (like `scipy.misc.imread`) but most people were in general -1 on this idea. I wasn't convinced at the moment but I've come around to the idea that `mne.io.read_raw_xxx` exists for a good reason. We can leave it in there to see how many people are using it as it's a thin wrapper over the `mne.io.read_raw_xxx` functions but I don't want to force people to use it. 

I've now added back the `hpi` parameter in the `write_raw_bids` function for now as otherwise there was this coupling between the two functions (`write_raw_bids` and `mne_bids.io.read_raw`) which was a bit odd. Might be worth waiting for @teonbrooks comments before merging this PR though as the generalized reader was his idea.

Reply:
The only reason we would need it here is because of the HPI issue with KIT data. If this functionality was moved into the MNE code then you could go back to to using the specific functions with no worries. If people think that the change made here is good then really I think it should be implemented in MNE that way there is no difference between KIT raw files in MNE-BIDS and MNE-python

Reply:
yes, there is probably something to fix upstream in MNE but so far it seems like an isolated case for KIT data. I can live with one extra parameter for that. We can deprecate it later if we come up with a better solution. But I think the raw files from `mne_bids.io.read_raw` and `mne.io.read_raw_xxx` should definitely be identical.